tests/MonitorEditControllerTest.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace Cronbeat\Tests;
+
+use Cronbeat\Controllers\MonitorController;
+use Cronbeat\RedirectException;
+use PHPUnit\Framework\Assert;
+
+class MonitorEditControllerTest extends DatabaseTestCase {
+    private ?MonitorController $controller = null;
+    private int $userId = 0;
+    private string $username = 'editor';
+    private string $passwordHash = '';
+
+    protected function setUp(): void {
+        parent::setUp();
+
+        $this->passwordHash = hash('sha256', 'pw');
+        $db = $this->getDatabase();
+        $db->createUser($this->username, $this->passwordHash);
+        $validated = $db->validateUser($this->username, $this->passwordHash);
+        if ($validated === false) {
+            throw new \RuntimeException('Failed to validate test user');
+        }
+        $this->userId = $validated;
+
+        $_SESSION = [];
+        $_GET = [];
+        $_POST = [];
+        $_SERVER['REQUEST_METHOD'] = 'GET';
+        $_SESSION['user_id'] = $this->userId;
+
+        $this->controller = new MonitorController($db);
+    }
+
+    protected function tearDown(): void {
+        $_SESSION = [];
+        $_GET = [];
+        $_POST = [];
+        parent::tearDown();
+    }
+
+    private function getController(): MonitorController {
+        if ($this->controller === null) {
+            throw new \RuntimeException('Controller not initialized');
+        }
+        return $this->controller;
+    }
+
+    public function testEditPageShowsFormAndDeleteLink(): void {
+        // Given
+        $db = $this->getDatabase();
+        $uuid = $db->createMonitor('Original Name', $this->userId);
+        if ($uuid === false) {
+            throw new \RuntimeException('Failed to create monitor for test');
+        }
+
+        // When
+        $_SERVER['REQUEST_URI'] = "/monitor/$uuid/edit";
+        $html = $this->getController()->doRouting();
+
+        // Then
+        Assert::assertStringContainsString('<form', $html);
+        Assert::assertStringContainsString('name="name"', $html);
+        Assert::assertStringContainsString('Original Name', $html);
+        // Delete button moved from dashboard to edit page
+        Assert::assertStringContainsString("/dashboard/delete/" . $uuid, $html);
+    }
```

Comment:
@jetbrains-junie this is two different requirements. split the test into "shows form" and "can delete"